packages/react-router/src/RouterProvider.tsx

```diff
@@ -43,8 +43,13 @@ export type NavigateFn<TRouteTree extends AnyRoute> = <
   opts: NavigateOptions<TRouteTree, TFrom, TTo, TMaskFrom, TMaskTo>,
 ) => Promise<void>
 
-export type BuildLocationFn<TRouteTree extends AnyRoute> = (
-  opts: ToOptions<TRouteTree> & {
+export type BuildLocationFn<TRouteTree extends AnyRoute> = <
+  TFrom extends RoutePaths<TRouteTree> | string = string,
```

Comment:
please use 
`TFrom extends RoutePaths<TRouteTree> | string = RoutePaths<TRouteTree>`

Reply:
Good call, I've just changed it.

But now honest question: why the behavior of `NavigationFn` isn't also like this?
It's `TFrom` default is `string` and in my mind it would also make sense to be `RoutePaths<TRouteTree>`

Reply:
yeah could be changed as well